src/auth/pluggableauthclient.ts

```diff
@@ -0,0 +1,253 @@
+   */
+  constructor(
+    options: PluggableAuthClientOptions,
+    additionalOptions?: RefreshOptions
+  ) {
+    super(options, additionalOptions);
+    this.command = options.credential_source.command;
+    if (!this.command) {
+      throw new Error('No valid Pluggable Auth "credential_source" provided.');
+    }
+    //Check if the provided timeout exists and if it is valid.
+    if (options.credential_source.timeout_millis === undefined) {
+      this.timeoutMillis = DEFAULT_EXECUTABLE_TIMEOUT_MILLIS;
+    } else {
+      this.timeoutMillis = options.credential_source.timeout_millis;
+      if (
+        this.timeoutMillis < MINIMUM_EXECUTABLE_TIMEOUT_MILLIS ||
+        this.timeoutMillis > MAXIMUM_EXECUTABLE_TIMEOUT_MILLIS
+      ) {
+        throw new Error(
+          `Timeout must be between ${MINIMUM_EXECUTABLE_TIMEOUT_MILLIS} and ` +
+            `${MAXIMUM_EXECUTABLE_TIMEOUT_MILLIS} milliseconds.`
+        );
+      }
+    }
+
+    this.outputFile = options.credential_source.output_file;
+  }
+
+  /**
+   * Triggered when an external subject token is needed to be exchanged for a
+   * GCP access token via GCP STS endpoint.
+   * This uses the `options.credential_source` object to figure out how
+   * to retrieve the token using the current environment. In this case,
+   * this calls a user provided executable which returns the subject token.
+   * The logic is summarized as:
+   * 1. Validated that the executable is allowed to run. The
+   *    GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES environment must be set to
+   *    1 for security reasons.
+   * 2. If an output file is specified by the user, check the file location
+   *    for a response. If the file exists and contains a valid response,
+   *    return the subject token from the file.
+   * 3. Call the provided executable and return response.
+   * @return A promise that resolves with the external subject token.
+   */
+  async retrieveSubjectToken(): Promise<string> {
+    // Check if the executable is allowed to run.
+    if (process.env[GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES] !== '1') {
+      throw new Error(
+        'Pluggable Auth executables need to be explicitly allowed to run by ' +
+          'setting the GOOGLE_EXTERNAL_ACCOUNT_ALLOW_EXECUTABLES environment ' +
+          'Variable to 1.'
+      );
+    }
+
+    let executableResponse: ExecutableResponse | undefined;
+    // Try to get cached executable response from output file
+    if (this.outputFile) {
+      executableResponse = await this.retrieveCachedResponse();
+    }
+
+    // If no response from output file, call the executable
+    if (!executableResponse) {
+      // Set up environment map with required values for the executable
+      const envMap = new Map();
+      envMap.set('GOOGLE_EXTERNAL_ACCOUNT_AUDIENCE', this.audience);
+      envMap.set('GOOGLE_EXTERNAL_ACCOUNT_TOKEN_TYPE', this.subjectTokenType);
+      // Set to 0 for workloadIdentityFederation
```

Comment:
// Interactive mode is not supported.

Reply:
FYI, comment is the same for the java PR, (says set to 0 for workload identity federation but doesn't mention why) I agree that the additional comment is good but should add it to the other repo PR's as well.